library/Fission/Mail/Class.hs

```diff
@@ -0,0 +1,11 @@
+module Fission.Mail.Class (MonadMail (..)) where
+
+import           Fission.Prelude
+import           Servant.Client
+
+import           Fission.Mail.Recipient.Types
+import           Fission.Mail.SendInBlue.Types 
+import           Fission.Challenge.Types
+
+class Monad m => MonadMail m where
+  sendVerifyEmail :: Recipient -> Challenge -> m (Either ClientError SendEmailRes)
```

Comment:
`SendEmailRes`

Actually type out `Response`. You can break that out into a submodule as well to compress the text contextually:

```haskell
import qualified Fission.Mail.SendInBlue.Types as Email

foo :: Email.Response
```

I'd also recommend adding an `E` to `Mail`. Unambiguous, and it's one character.

Reply:
sounds good :+1: